app/customer.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import math
+
+from app.car import Car
+from app.shop import Shop
+
+
+class Customer:
+    def __init__(self, customer_data: dict) -> None:
+        self.name = customer_data.get("name")
+        self.product_cart: dict = customer_data.get("product_cart")
+        self.location: list[int] = customer_data.get("location")
+        self.money: float | int = customer_data.get("money")
+        self.car: Car = Car(customer_data.get("car"))
+
+    @staticmethod
+    def calculate_distance(
+            home_location: list[int],
+            shop_location: list[int]
+    ) -> float:
+        return math.sqrt(
+            (home_location[0] - shop_location[0]) ** 2
+            + (home_location[1] - shop_location[1]) ** 2
+        ) * 2
+
+    def calculate_fuel_cost(
+            self,
+            fuel_price: float,
+            shop_location: list[int]
+    ) -> float:
+        distance = self.calculate_distance(self.location, shop_location)
+        return distance * self.car.fuel_consumption * fuel_price / 100
+
+    def get_trip_price(
+            self,
+            fuel_price: float,
+            shop_location: list,
+            shop_products: dict
+    ) -> float:
+        fuel_cost = self.calculate_fuel_cost(fuel_price, shop_location)
+        products_cost = sum(
+            count * shop_products[product]
+            for product, count in self.product_cart.items()
+            if product in shop_products

```

Comment:
Ensure that the `shop_products` dictionary contains all products in the `product_cart`. If a product is missing, it might lead to a `KeyError`. Consider using a default value or handling the exception.